plugins/dev-create/bin/plugin.js

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env node
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A script for creating Blockly plugins based on pre-existing
+ * templates.
+ */
+
+'use strict';
+
+const chalk = require('chalk');
+const fs = require('fs-extra');
+const path = require('path');
+const execSync = require('child_process').execSync;
+const {createDir} = require('./common');
+
+const pluginTypes = ['plugin', 'field', 'block', 'theme'];
+
+const root = process.cwd();
+
+exports.createPlugin = function(pluginName, options) {
+  let gitRoot = '';
+  let gitURL = '';
+  try {
+    gitRoot = execSync(`git rev-parse --show-toplevel`).toString().trim();
+    gitURL = execSync(`git config --get remote.origin.url`)
+        .toString()
+        .trim()
+        .replace(/\.git$/, '');
+  } catch (err) {
+  // NOP
+  }
+
+  const isGit = !!gitURL;
+  const isFirstParty = options.firstParty || gitURL == 'https://github.com/google/blockly-samples';
+
+  /**
+   * Gets the name of the plugin prefixed with the type.
+   * Default to the plugin name if the type=plugin, otherwise use [type]-[name].
+   * @param {string} name Plugin name.
+   * @param {string} type Plugin type.
+   * @return {string} Plugin name prefixed with type.
+   */
+  const getPrefixedName = function(name, type) {
+  // Don't add 'plugin' prefix for default type
+    if (type == 'plugin') return name;
+    // If the name is already prefixed, just return the name.
+    if (name.startsWith(`${type}-`)) return name;
+    // Add the prefix if not already present.
+    return `${type}-${name}`;
+  };
+
+  // Default to type=plugin.
+  const pluginType = options.type || 'plugin';
+  const prefixedName = getPrefixedName(pluginName, pluginType);
+  const pluginDir = options.dir || prefixedName;
+  const pluginPath = path.join(root, pluginDir);
+  const pluginAuthor = options.author || (isFirstParty ? 'Blockly Team' : '');
+
+  const isTypescript = options.typescript;
+  const skipInstall = options.skipInstall;
+
+  // Check plugin type.
+  if (!pluginTypes.includes(pluginType)) {
+    console.error(`Unknown plugin type: ${chalk.red(pluginType)}`);
+    console.log(`Available types: ${chalk.green(pluginTypes.join(', '))}`);
+    process.exit(1);
+  }
+
+  // Create the new directory if needed.
+  createDir(pluginPath);
+
+  console.log(`Creating a new Blockly\
+ ${chalk.green(pluginType)} with name ${chalk.green(pluginName)}\
+ in ${chalk.green(pluginPath)}.\n`);
```

Comment:
[No line continuations](go/tsstyle#syntax-no-line-continuations):

```suggestion
  console.log(
      `Creating a new Blockly ${chalk.green(pluginType)} with name ` +
      `${chalk.green(pluginName)} in ${chalk.green(pluginPath)}.\n`);
```